### PR TITLE
Add missing function name

### LINKED ISSUE #3613: cli cause AbiFunctionNotFoundError: Function not found on ABI.
### Describe the bug

when using writeHook generated from wagmi cli, it will cause 
```
AbiFunctionNotFoundError: Function not found on ABI.
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/new-wagmi-hhmcve?file=src%2FApp.tsx

### Steps To Reproduce

connect and click write

### Wagmi Version

2.5.7

### Viem Version

2.7.10

### TypeScript Version

_No response_

### Check existing issues

- [X] I checked there isn't [already an issue](https://github.com/wevm/wagmi/issues) for the bug I encountered.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by m1heng**
fix #3613

### CODE REVIEW

**Review by tmm — CHANGES_REQUESTED** [review]
Looks great! Can you create a patch changeset with `pnpm changeset`?

**Comment by m1heng** [comment]
changeset added

**Comment by m1heng** [comment]
@tmm maybe we can have this merged earlier so that I can remove the patch in my private repo

**Comment by jojoo-eth** [comment]
Hello, I'm troubled by this issue. May I ask if it can be merged?

**Comment by kschrute** [comment]
I'd appreciate it if this was merged as well.

**Comment by kschrute** [comment]
@tmm

File: .changeset/tiny-bees-own.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"wagmi": patch
+---
+
+Add missing function name in codegen function
```
Comment (tmm) [inline .changeset/tiny-bees-own.md:5]:
```suggestion
Added missing `functionName` parameter to `useWriteContract` codegen helper.
```

**Review by tmm — APPROVED** [review]

**Pull request merged**